File: backend/src/main/java/com/wavii/model/enums/Subscription.java
package com.wavii.model.enums;

/**
 * Enum que define los planes de suscripcion disponibles.
 */
public enum Subscription {
    /** Plan gratuito con funcionalidades limitadas */
    FREE,
    /** Plan de pago con acceso completo a tablaturas y desafios */
    PLUS,
    /** Plan educativo para instituciones y escuelas */
    SCHOLAR;

    public static Subscription fromDatabaseValue(String value) {
        if (value == null || value.isBlank()) {
            return FREE;
        }

        if ("EDUCATION".equalsIgnoreCase(value) || "SCHOLAR".equalsIgnoreCase(value)) {
            return SCHOLAR;
        }

        return Subscription.valueOf(value.toUpperCase());
    }

    public String toDatabaseValue() {
        return this == SCHOLAR ? "EDUCATION" : name();
    }

    public String toPublicId() {
        return this == SCHOLAR ? "education" : name().toLowerCase();
    }
}

File: frontend/src/utils/subscription.ts
export type NormalizedSubscription = 'free' | 'plus' | 'education';

export function normalizeSubscription(value: string | null | undefined): NormalizedSubscription {
  const normalized = String(value ?? '').trim().toLowerCase();

  if (normalized === 'plus') {
    return 'plus';
  }

  if (normalized === 'education' || normalized === 'scholar') {
    return 'education';
  }

  return 'free';
}

export function hasScholarAccess(value: string | null | undefined): boolean {
  return normalizeSubscription(value) === 'education';
}
